Lib/_collections_abc.py

```diff
@@ -409,6 +415,81 @@ def __subclasshook__(cls, C):
         return NotImplemented
 
 
+class _CallableGenericAlias(GenericAlias):
+    """ Internal class specifically for consistency between the ``__args__`` of
+    ``collections.abc.Callable``'s and ``typing.Callable``'s ``GenericAlias``.
+
+    See :issue:`42195`.
+    """
+
+    def __new__(cls, *args, **kwargs):
+        if not isinstance(args, tuple) or len(args) != 2:
+            raise TypeError("Callable must be used as "
+                            "Callable[[arg, ...], result].")
+        _typ, _args = args
+        if not isinstance(_args, (tuple, EllipsisType)) or len(_args) != 2:
+            raise TypeError("Callable must be used as "
+                            "Callable[[arg, ...], result].")
+        t_args, t_result = _args
+        if not isinstance(t_args, (list, EllipsisType)):
+            raise TypeError("Callable[args, result]: args must be a list. Got"
+                            f" {_type_repr(t_args)}")
+
+        ga_args = []
+        for arg in _args:
+            if isinstance(arg, list):
+                ga_args.extend(arg)
+            else:
+                ga_args.append(arg)
+        return super().__new__(cls, _typ, tuple(ga_args))
+
+    def __init__(self, *args, **kwargs):
+        pass
+
+    def __eq__(self, other):
+        if not isinstance(other, GenericAlias):
+            return NotImplemented
+        return (self.__origin__ == other.__origin__
+                and self.__args__ == other.__args__)
+
+    def __hash__(self):
+        return super().__hash__()
+
+    def __repr__(self):
+        t_args = self.__args__[:-1]
+        t_result = self.__args__[-1]
+
+        orig = _type_repr(self.__origin__)
+        args = f"{', '.join(_type_repr(a) for a in t_args)}"
+        result = _type_repr(t_result)
+
+        if not isinstance(t_args[0], EllipsisType):
+            args = f"[{args}]"
+
+        return f"{orig}[{args}, {result}]"
+
+
+def _type_repr(obj):
+    """Return the repr() of an object, special-casing types (internal helper).
+
+    If obj is a type, we return a shorter version than the default
+    type.__repr__, based on the module and qualified name, which is
+    typically enough to uniquely identify a type.  For everything
+    else, we fall back on repr(obj).
+
+    Borrowed from :mod:`typing`.
```

Comment:
Explain you can't import it from there because typing shouldn't be a dependency of this module. You don't need to copy the full docstring IMO.